indica/converters/abstractconverter.py

```diff
@@ -310,3 +371,168 @@ def encode(self) -> str:
     def decode(json: str) -> "CoordinateTransform":
         """Takes some JSON and decodes it into a CoordinateTransform object."""
         pass
+
+    def get_machine_boundaries(
+        self,
+        machine_dimensions: Tuple[Tuple[float, float], Tuple[float, float]] = (
+            (1.83, 3.9),
+            (-1.75, 2.0),
+        ),
+        npts: int = 1000,
+    ) -> Tuple[dict, ArrayLike]:
+        angles = np.linspace(0.0, 2 * np.pi, npts)
+        x_wall_inner = machine_dimensions[0][0] * np.cos(angles)
+        x_wall_outer = machine_dimensions[0][1] * np.cos(angles)
+        y_wall_inner = machine_dimensions[0][0] * np.sin(angles)
+        y_wall_outer = machine_dimensions[0][1] * np.sin(angles)
+        z_wall_lower = machine_dimensions[1][0]
+        z_wall_upper = machine_dimensions[1][1]
+
+        boundaries = {
+            "x_in": x_wall_inner,
+            "x_out": x_wall_outer,
+            "y_in": y_wall_inner,
+            "y_out": y_wall_outer,
+            "z_up": z_wall_upper,
+            "z_low": z_wall_lower,
+        }
+
+        return boundaries, angles
+
+    def get_equilibrium_boundaries(
+        self, tplot: float, npts: int = 1000
+    ) -> Tuple[dict, ArrayLike, DataArray]:
+
+        boundaries = {}
+        angles = np.linspace(0.0, 2 * np.pi, npts)
+        if hasattr(self, "equilibrium"):
+            angles = np.linspace(0.0, 2 * np.pi, npts)
+            rho_equil = self.equilibrium.rho.sel(t=tplot, method="nearest")
+            rho_equil = xr.where(rho_equil < 1.05, rho_equil, np.nan)
+            core_ind = np.where(np.isfinite(rho_equil.interp(z=0)))[0]
+            R_lfs = rho_equil.R[core_ind[0]].values
+            R_hfs = rho_equil.R[core_ind[-1]].values
+            x_plasma_inner = R_hfs * np.cos(angles)
+            x_plasma_outer = R_lfs * np.cos(angles)
+            y_plasma_inner = R_hfs * np.sin(angles)
+            y_plasma_outer = R_lfs * np.sin(angles)
+
+            boundaries = {
+                "x_in": x_plasma_inner,
+                "x_out": x_plasma_outer,
+                "y_in": y_plasma_inner,
+                "y_out": y_plasma_outer,
+            }
+        return boundaries, angles, rho_equil
+
+
+def find_wall_intersections(
```

Comment:
Any reason for the `origin` and `direction` arguments to be type `Tuple`? `LabeledArray` or `ArrayLike` might be better.